### PR TITLE
6 And 7 Feb MIS_Automation

### PR DESCRIPTION
**Pull request by Gem-Shrutissingh**
@gem-rishabhrastogi

### CODE REVIEW

File: src/main/java/com/qa/MIS_Automation/StepDefination/StepDefination.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        DriverAction.click(Locators.formType(formType));
+        DriverAction.waitSec(3);
+        DriverAction.fileUpload(Locators.chooseFile, "C:\\Users\\sh.singh9\\Desktop\\MIS GemJar\\MIS_Automation_GemJar - Final\\src\\main\\resources\\17 may.docx");
+        DriverAction.waitSec(4);
+    }
+
+    @Then("Verify the {string}")
+    public void verifyThe(String warningMsg) {
+        DriverAction.waitUntilElementAppear(Locators.warningMsg,5);
+        System.out.println("Error msg appeared");
```
Comment (gem-rishabhrastogi) [inline src/main/java/com/qa/MIS_Automation/StepDefination/StepDefination.java:187]: Replace System.out.print statements with Reporting functionalities

File: src/main/java/com/qa/MIS_Automation/StepDefination/StepDefination.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    }
+
+    @Then("User clicks on okbtn")
+    public void userClicksOnOkbtn() {
+        DriverAction.waitUntilElementAppear(Locators.okBtn,4);
+        DriverAction.click(Locators.okBtn);
```
Comment (gem-rishabhrastogi) [inline src/main/java/com/qa/MIS_Automation/StepDefination/StepDefination.java:193]: Before clicking an element, either enclose it within try-catch block or validate the existence of element.

File: src/main/resources/features/MIS_Automation.feature [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    Given User enters Username as "shruti.singh"
+    And User enters Password as "Gemini@123"
+    When User clicks on sign in
+    And User Verifies Landing Page
+    Then User clicks on closeBtn
+
+
+  Scenario Outline: Open View Forms Page
+
+    Given User clicks on tab "<parent Tab>" and "<child Tab>"
```
Comment (gem-rishabhrastogi) [inline src/main/resources/features/MIS_Automation.feature:13]: Refrain from using spaces between argument names. For example, replace "parent Tab" with "parentTab"

File: src/main/resources/features/MIS_Automation.feature [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    Then User verifies element "<FirstElement>"
```
Comment (gem-rishabhrastogi) [inline src/main/resources/features/MIS_Automation.feature:14]: The first character of all the arguments should be in small case. Replace "FirstElement" with "firstElement".

**Review by gem-rishabhrastogi — CHANGES_REQUESTED** [review]
I've added some suggestions in a couple of files. Please work on those.

**Comment by Gem-Shrutissingh** [comment]
Hi Rishabh ;

Thanks for the review will work on that.


Thanks & Regards,
Shruti Singh

Software Engineer - L1
M: +91 7895909198

Quality Assurance Department


[cid:9e47948a-cda9-46a7-8ea5-4bd09295f615]

LinkedIn<https://www.linkedin.com/company/gemini-solutions-india/mycompany/?viewAsMember=true> | Facebook<https://www.facebook.com/gemsolindia> | Instagram<https://www.instagram.com/gemini.solutions/> | Glassdoor<https://www.glassdoor.co.in/Overview/Working-at-Gemini-Solutions-EI_IE1333027.11,27.htm>


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Review by gem-rishabhrastogi — APPROVED** [review]
Good to go @gemini-akashGarg

**Pull request merged**